core/config/v2/types.go

```diff
@@ -1002,3 +1003,31 @@ func (ins *Insecure) setFrom(f *Insecure) {
 		ins.OCRDevelopmentMode = f.OCRDevelopmentMode
 	}
 }
+
+type MercuryCredentials struct {
+	URL      *models.SecretURL
+	Username *models.Secret
+	Password *models.Secret
+}
+
+type MercurySecrets struct {
+	Credentials map[string]MercuryCredentials
+}
+
+func (m *MercurySecrets) ValidateConfig() (err error) {
+	urls := make(map[string]struct{}, len(m.Credentials))
+	for name, creds := range m.Credentials {
+		if name == "" {
+			return errors.New("`Name` must be set for all mercury credentials")
+		}
+		if creds.URL == nil {
+			return errors.New("`url` must be set for all mercury credentials")
+		}
+		s := creds.URL.String()
+		if _, exists := urls[s]; exists {
+			return errors.New("credentials: may not contain duplicate URLs")
```

Comment:
Please use the standard errors and helpers:
- `v2.NewErrDuplicate()`
- `v2.ErrMissing`
- `v2.ErrEmpty`